spec/rubocop/ast/masgn_node_spec.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::AST::MasgnNode do
+  let(:masgn_node) { parse_source(source).ast }
+  let(:source) { 'x, y = z' }
+
+  describe '.new' do
+    context 'with a `masgn` node' do
+      it { expect(masgn_node).to be_a(described_class) }
+    end
+  end
+
+  describe '#names' do
+    subject { masgn_node.names }
+
+    let(:source) { 'a, @b, @@c, $d, E, *f = z' }
+
+    it { is_expected.to eq(%i[a @b @@c $d E f]) }
+
+    context 'with nested `mlhs` nodes' do
+      let(:source) { 'a, (b, c) = z' }
+
+      it { is_expected.to eq(%i[a b c]) }
+    end
+
+    context 'with assignment inside `send` node on LHS' do
+      let(:source) { 'a, b[c+=1] = z' }
+
+      it { is_expected.to eq([:a, :'b[]']) } # rubocop:disable Style/SymbolArray
```

Comment:
Explicitly disabled `Style/SymbolArray` here because I thought this was better than `%i[a, b\[\]]`